src/Compilers/Core/Portable/CodeGen/TokenMap.cs

```diff
@@ -24,74 +23,91 @@ namespace Microsoft.CodeAnalysis.CodeGen
     /// </summary>
     internal sealed class TokenMap
     {
-        private readonly ConcurrentDictionary<object, uint> _itemIdentityToToken = new ConcurrentDictionary<object, uint>(ReferenceEqualityComparer.Instance);
+        private readonly ConcurrentDictionary<IReferenceOrISignature, uint> _itemIdentityToToken = new ConcurrentDictionary<IReferenceOrISignature, uint>();
```

Comment:
nit: **If** C# 9.0 features are available to use here, I'd use target-typed new.

```suggestion
        private readonly ConcurrentDictionary<IReferenceOrISignature, uint> _itemIdentityToToken = new();
```

Reply:
Looks to be